deps/compiler-rt.mk

```diff
@@ -0,0 +1,144 @@
+##
+# In order to support fallbacks within llvm we need to support
+# compiler-rt. This means linking sys.so against it and resolving
+# symbols at during JIT. For the latter part we need to create
+# a .so that we can load, but compiler-rt only comes in a .a.
+#
+# There are several configurations to take into account.
+# 1. USE_SYSTEM_LLVM == 1
+#    libclang_rt.builtins is distributed with clang and so
+#    we assume that USE_SYSTEM_LLVM == 1 means that clang is also
```

Comment:
Yes this adds an implicit dependency on clang...
We could relax this a bit by using a standalone build for `USE_SYSTEM_LLVM == 1 && BUILD_COMPILER_RT == 1`, but for `USE_SYSTEM_LLVM == 1 && BUILD_COMPILER_RT == 0`, we need to fall back onto a system version of compiler-rt and that comes at least on Archlinux with clang.